### PR TITLE
feat: integrate sns

### PR DESCRIPTION
**Pull request by aganglada**
* Integrate SNS in `onNameLookup`
* Resolve domains on the Send flow

Screenshot 2025-07-01 at 15 47 15
Screenshot 2025-07-01 at 15 47 26

Not in scope:
* Showing `.sol` domain in confirmations (There is a Snap UI limitation right now)

### CODE REVIEW

File: packages/snap/src/core/handlers/onNameLookup/structs.ts [hunk 1]
```diff
@@ -0,0 +1,9 @@
+import { object, optional, string } from '@metamask/superstruct';
+
+import { NetworkStruct } from '../../validation/structs';
+
+export const SolanaNameLookupRequesstStruct = object({
```
Comment (xavier-brochard) [inline packages/snap/src/core/handlers/onNameLookup/structs.ts:5]: Double "s" in "Requesst"

File: packages/snap/src/features/send/utils/isSolanaDomain.test.ts [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+  describe('when given edge cases', () => {
+    it('returns false for null input', () => {
+      expect(isSolanaDomain(null)).toBe(false);
+    });
+
+    it('returns false for empty string', () => {
+      expect(isSolanaDomain('')).toBe(false);
+    });
+
+    it('returns true for string with only .sol', () => {
```
Comment (xavier-brochard) [inline packages/snap/src/features/send/utils/isSolanaDomain.test.ts:53]: Should it return true for this case? It would assume that `.sol` is an invalid domain.

Comment (aganglada) [reply packages/snap/src/features/send/utils/isSolanaDomain.test.ts:53]: TRUE

**Review by ulissesferreira — APPROVED** [review]

**Review by ulissesferreira — APPROVED** [review]

**Review by ulissesferreira — APPROVED** [review]
